xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp

```diff
@@ -268,15 +268,20 @@ bool CDVDInputStreamPVRManager::PrevChannel(bool preview/* = false*/)
 bool CDVDInputStreamPVRManager::SelectChannelByNumber(unsigned int iChannelNumber)
 {
   PVR_CLIENT client;
+  CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
+  CFileItemPtr item(g_PVRChannelGroups->Get(currentChannel->IsRadio())->GetSelectedGroup()->GetByChannelNumber(iChannelNumber));
```

Comment:
if we select a channel by id via json-rpc we also need to change the group to one of the groups the channel lies in. The playing group and selected group is currently the same.